pkg/config/setup/config.go

```diff
@@ -973,6 +973,9 @@ func agent(config pkgconfigmodel.Config) {
 	config.SetKnown("proxy.http")
 	config.SetKnown("proxy.https")
 	config.SetKnown("proxy.no_proxy")
+
+	// Define the duration after which subagent configuration fetch will timeout
+	config.BindEnvAndSetDefault("subagent_config_fetch_timeout", "30s")
```

Comment:
I just remembered that there is already a config option somewhat for this: `server_timeout`
https://github.com/DataDog/datadog-agent/blob/f6d0c587696f58748dc04a5d4fd7ec54ee235f23/pkg/config/config_template.yaml#L749-L753

It's a timeout in the server, but IMO it makes sense to use it when querying the server too

Reply:
Done in fbaf2367b85c949e436f2e113abc67c570f018d5.